src/main/java/graphql/schema/DataFetchingFieldSelectionSetImpl.java

```diff
@@ -14,8 +15,9 @@
     private final static DataFetchingFieldSelectionSet NOOP = Collections::emptyMap;
 
     public static DataFetchingFieldSelectionSet newCollector(ExecutionContext executionContext, GraphQLType fieldType, List<Field> fields) {
-        if (fieldType instanceof GraphQLObjectType) {
-            return new DataFetchingFieldSelectionSetImpl(executionContext, (GraphQLObjectType) fieldType, fields);
+        GraphQLType unwrappedType = ExecutionTypeInfo.unwrapBaseType(fieldType);
+        if (unwrappedType instanceof GraphQLFieldsContainer) {
```

Comment:
right its because we want selection sets on Interfaces as well.  Otherwise anything that is described by an Interface will not be seen by the DataFetchingFieldSelectionSetImpl and FieldCollector